metis/lib/query_builder.rb

```diff
@@ -0,0 +1,66 @@
+require 'date'
+
+class Metis
+  class QueryBuilder
+
+    def initialize(base_query, params)
+      @base_query = base_query
+      @params = params
+    end
+
+    def build
+      @params.each do |param|
+        next unless param.key?(:attribute) && param.key?(:predicate) && param.key?(:value)
+
+        # Modify query behavior based on "type" of attribute
+        case param[:attribute]
+        when 'created_at', 'updated_at'
```

Comment:
Are there other things that would be useful to search on?

Reply:
What about "globs" searches that would allow parent folder filtering too?
For instance, my ide has two types of name searching

`abc.def` => matches any filename like "vabc.def" or "abc.defg"
`make/abc.def` matches any filename like "abc.def" but also within any directory structure containing "make"

I imagine wanting to, from the top of a data bucket, search for like "processed/HS103" to list all processed records in any structure for HS103

Reply:
Ah, that would be neat. Let me look into that.